src/App.tsx

```diff
@@ -1,29 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    if (currentPage !== 1) {
+      setCurrentPage(1);
+    }
+  };
+
+  const handleChangePage = (page: number) => {
+    setCurrentPage(page);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {currentPage * perPage - perPage + 1} -{' '}
+        {currentPage * perPage > items.length ? 42 : currentPage * perPage} of
+        42)
```

Comment:
The number '42' is hardcoded here, which might not be ideal if the number of items changes. Consider using `items.length` instead to dynamically reflect the total number of items.